### PR TITLE
implements Twin design pattern #63

### PR DESCRIPTION
**Pull request by hoswey**
please help review.

### CODE REVIEW

**Comment by hoswey** [comment]
the converages check is so strange as I find my source at lease reach to 86.36.

alt text

**Comment by iluwatar** [comment]
- [ ] Put `under review` badge to the pull request
- [ ] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [ ] Is the example code commented well enough?
- [ ] Is the example code following JavaDoc conventions?
- [ ] Are the project [coding conventions](https://github.com/iluwatar/java-design-patterns/wiki/3.-Coding-conventions) being followed?
- [ ] Is the class diagram generated correctly?
- [ ] Is the `index.md` implemented correctly so the pattern will show correctly on the web site?
- [ ] Is the originator of the design pattern properly credited?

File: twin/src/main/java/com/iluwatar/twin/App.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * Twin pattern is a design pattern which provides a standard solution to simulate multiple
+ * inheritance in java.
+ * 
+ * <p>
+ * In this example, there is a ball game, a ball needs to subclass {@link GameItem} which provide
+ * some common method like draw and click. Moreover, it needs to subclass {@link Thread} as ball is
+ * a moving item (we use {@link Thread} instead of implements {@link Runnable} for example only)
+ * <p>
+ * Threre is scenario, when user click the ball, the ball will stop, when user click it gain, it
```
Comment (iluwatar) [inline twin/src/main/java/com/iluwatar/twin/App.java]: Typo "Threre"

File: twin/src/main/java/com/iluwatar/twin/App.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * will resume to move. We create two class, ons is {@link BallItem} which subclass {@link GameItem}
+ * , another is {@link BallThread} which subclass {@link Thread}. These two object both hold a field
+ * named "Twin" reference to another object. In {@link BallItem#click()}, it will invoke
+ * {@link BallThread} to suspend or resume moving; in {@link BallThread#run()}, it will invoke
+ * {@link BallItem} for drawing.
```
Comment (iluwatar) [inline twin/src/main/java/com/iluwatar/twin/App.java]: Multiple typos and unclear text. This should be rephrased.

**Comment by iluwatar** [comment]
Overall the implementation is well done and only minor issues were found. Please comment on this thread when you've implemented the changes and I will review again.

**Comment by hoswey** [comment]
Thanks @iluwatar 's comments. Fix is done, please review.

**Comment by iluwatar** [comment]
Thank you @hoswey Well done!

**Comment by npathai** [comment]
Thanks for your contribution @hoswey

**Pull request merged**